portfolio/src/main/webapp/script.js

```diff
@@ -27,12 +27,39 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-
 async function showServerTime() {
     const responseFromServer = await fetch('/hello');
-    const myObject = await responseFromServer.json();
-    const dateContainer = document.getElementById('m-container');
-    const single_message = myObject[Math.floor(Math.random() * myObject.length)];
-    dateContainer.innerText = single_message;
+    const response = await responseFromServer.json();
+    const messagecontainer = document.getElementById('m-container');
+    if (response["comments"].length == 0) {
+        messagecontainer.innerHTML = "No comments";
+    } else {
+        messagecontainer.innerHTML = response["comments"][Math.floor(Math.random() * response["comments"].length)];
+    }
 }
 
+async function showTranslation(){
+
+    const text = document.getElementById('text').value;
+    const languageCode = document.getElementById('language').value;
+
+    const resultContainer = document.getElementById('result');
+    resultContainer.innerText = 'Loading...';
+
+    //appends text and language code key/value pair to the URLSearchParams object
+    const params = new URLSearchParams();
+    params.append('text', text);
+    params.append('languageCode', languageCode);
+
+    fetch('/translate', {
+      method: 'POST',
+      body: params
+    }).then(response => response.text()) //read the response
+    .then((translatedMessage) => { 
+      resultContainer.innerText = translatedMessage;
+    });
+  }
```

Comment:
The changes seem to be for this week's walkthrough? Please make this a different PR. A set up could be:

```
                                 | Feature2 branch (PR 2)
                                 |
                                 |
             ------------|
             |
             |
   ------|
   |.      feature1 branch (this PR)
   |
   |
master
```

Reply:
https://www.youtube.com/watch?v=a2OSRBz107I we can go over this together if you want